src/redux/data.js

```diff
@@ -198,6 +201,27 @@ const formatErrors = (errors) => {
   }
 };
 
+/**
+ * Helper for calculating the difference between old and new state
+ * with model.save(). Borrows heavily from Ember Data.
+ * @param {Object} oldData - current state of attributes in store
+ * @param {Object} newData - requested new state of attributes
+ * @returns {Object} set of attributes with change
+ */
+const getChangedAttributes = (oldData, newData) => {
+  let diffData = Object.create(null);
+  let newDataKeys = Object.keys(newData);
+
+  for (let i = 0, length = newDataKeys.length; i < length; i++) {
+    let key = newDataKeys[i];
+    if (oldData[key] !== newData[key]) {
+      diffData[key] = [oldData[key], newData[key]];
```

Comment:
i dunno how i feel about relying on array position to denote something qualitative about data we're storing

Reply:
That's a good point. It was the approach taken in Ember Data's `changedAttributes()`, but that doesn't mean we have to do it identically. Is this preferable?
```
changedAttributes: {
  selectedCount: {
    old: 7,
    new: 0
  ],
  isSelected: [
    old: true, 
    new: false
  ]
}
```

Reply:
to me, yes.  but i'm open to discussion on it.

Reply:
I like the array notation because it's simple and most of this data stuff is inspired by Ember.

I'm also open to the object notation, as it's a bit more clear. But I'd recommend something like `prev` and `next` since `new` might not be the actual value of the attribute on the record. Like if the record changed elsewhere and is re-requested, `model.isSelected` might not reflect `model.update.changedAttributes.isSelected.new`.

Reply:
i agree with the `prev` and `next` comment if we go the object route.  i'm down for the array notation so long as we make a comment for future travelers.  it should be pretty clear by the code, especially if this is like a well known facet of ember-data (which i don't really have the context to know), but worth at least a note imo